### PR TITLE
Add nRF51 SPI master driver

### PR DESCRIPTION
**Pull request by Fabien-Chouteau**
And other related changes:
 - MicroBit SPI support
 - MicroBit I2C device definition
 - ST7735R (SPI screen): improve drawing performances in buffer-less mode*

### CODE REVIEW

File: arch/ARM/Nordic/devices/nrf51-device.ads [hunk 1]
```diff
@@ -76,9 +77,20 @@ package nRF51.Device is
    RTC_0 : aliased Real_Time_Counter (NRF51_SVD.RTC.RTC0_Periph'Access);
    RTC_1 : aliased Real_Time_Counter (NRF51_SVD.RTC.RTC1_Periph'Access);
 
+
+   --  Be carefull of shared ressources between the TWI and SPI controllers.
```
Comment (pat-rogers) [inline arch/ARM/Nordic/devices/nrf51-device.ads]: ressources  => resources

File: arch/ARM/Nordic/drivers/nrf51-spi_master.adb [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+--   (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE  --
+--   OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.   --
+--                                                                          --
+------------------------------------------------------------------------------
+
+with NRF51_SVD.SPI; use NRF51_SVD.SPI;
+
+package body nRF51.SPI_Master is
+
+   function Transfer (This     : in out SPI_Master;
```
[the remark this answers is not in the document]
Comment (Fabien-Chouteau) [reply arch/ARM/Nordic/drivers/nrf51-spi_master.adb]: I'll change it to a procedure.

[the remark this answers is not in the document]
Comment (pat-rogers) [reply arch/ARM/Nordic/drivers/nrf51-spi_master.adb]: That's probably best, I agree. (Thanks!)

**Comment by rtyler** [comment]
This is really exciting @Fabien-Chouteau! Thanks for taking the time to add this support

**Pull request merged**